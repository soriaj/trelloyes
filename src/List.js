import React from 'react';
import Card from './Card'
import './List.css'

function List(props){
   // const headers = props.header;
   // const cardIds = props.cards;
   return (
      <section className='List'>
         <header className='List-header'>
            <h3>{props.header}</h3>
         </header>
         <div className='List-cards'>
            {props.cards.map(cur => 
               <Card 
                  key={cur.id}
                  title={cur.title}
                  content={cur.content}
               />
            )}
            <button type='button' className='List-add-button'>
               +Add More Stuff
            </button>
         </div>
      </section>
   )
}

export default List;