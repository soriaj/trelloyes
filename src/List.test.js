import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';
import List from './List';

describe('Lists component', () => {
   const myList = [{
         id: '1',
         header: 'First list',
         cardIds: [ 'a', 'b', 'e', 'f', 'g', 'j', 'l', 'm' ]
   }];

   const myTestCards = 
      [
         { id: 'a', title: 'First card', content: 'lorem ipsum' },
         { id: 'b', title: 'Second card', content: 'lorem ipsum' },
         { id: 'c', title: 'Third card', content: 'lorem ipsum' }
      ];

   it('renders UI without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<List cards={myTestCards} />, div);
      ReactDOM.unmountComponentAtNode(div);
   });

   // it('renders the UI as expected with Snapshot', () => {
	// 	const tree = TestRenderer.create(<List header={myList.header} cards={myTestCards} />).toJSON();
	// 	expect(tree).toMatchSnapshot();
	// });
});