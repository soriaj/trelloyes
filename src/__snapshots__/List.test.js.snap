// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lists component renders the UI as expected with Snapshot 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h3 />
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        Second card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        Third card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      +Add More Stuff
    </button>
  </div>
</section>
`;
